src/OpenTelemetry.DynamicActivityBinding/public/ActivityStub.cs

```diff
@@ -0,0 +1,580 @@
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.DynamicActivityBinding
+{
+    /// <summary>
+    /// This is the type used by auto-instrumentation code to generate trace data. It is based
+    /// on the <c>System.Diagnostics.Activity</c> type. The latter can't be used directly by the
+    /// instrumentation code since it isn't feasible to determine prior runtime if it is going
+    /// to be possible to load it and if possible the exact version that is going to be loaded.
+    /// </summary>
+    public struct ActivityStub : IDisposable
+    {
+        // Temporary constant to be used with NotImplementedException.
+        // TODO: remove it when functionatily is implemented.
+        private const string MissingCapability = "Currently ActivityStub only supports no-op";
+
+        private ActivityStub(object? activityInstance)
+        {
+            TraceId = null;
+            SpanId = null;
+            ParentSpanId = null;
+            ForceDefaultIdFormat = false;
+
+            ActivityInstance = activityInstance;
+        }
+
+        /// <summary>
+        /// Gets the current ActivityStub for the current thread. This flows across async calls.
+        /// </summary>
+        /// <remarks>
+        /// Unlike <c>System.Diagnostics.Activity</c> ActivityStub is a struct so <c>null</c> can't be
+        /// used to indicate "no current activity".
+        /// <para/>
+        /// Provides functionality of <c>System.Diagnostics.DiagnosticSource.Activity.Current</c>.
+        /// Introduction: System.Diagnostics.DiagnosticSource, Version=4.0.2.1
+        /// </remarks>
+        public static ActivityStub Current
+        {
+            get
+            {
+                if (!DynamicLoader.EnsureInitialized())
+                {
+                    // TODO: is the no-op stub enough to represent "no current activity"? no need to differentiate
```

Comment:
Is this `TODO` something we want to start discussing in this PR, or is it a reminder for a future discussion?

Reply:
I would like to follow it up as a separate discussion.